android/app/src/main/java/com/mattermost/helpers/PushNotificationDataHelper.kt

```diff
@@ -136,20 +136,24 @@ class PushNotificationDataRunnable {
                 }
                 for(i in 0 until it.size()) {
                     val thread = it.getMap(i)
-                    val threadId = thread.getString("id")
+                    val threadId = thread?.getString("id")
                     if (threadId != null) {
                         if (threadIds.contains(threadId)) {
                          // replace the values for participants and is_following
                             val index = threadsArray.indexOfFirst { el -> el.getString("id") == threadId }
                             val prev = threadsArray[index]
                             val merge = Arguments.createMap()
                             merge.merge(prev)
-                            merge.putBoolean("is_following", thread.getBoolean("is_following"))
-                            merge.putArray("participants", thread.getArray("participants"))
+                            thread?.let { t ->
```

Comment:
I guess this can be moved up before getting the id

Reply:
Moved.